hash_table/[M]970.cpp

```diff
@@ -0,0 +1,45 @@
+// Problem: 970. Powerful Integers - Medium
+// Link: https://leetcode.com/problems/powerful-integers/
+
+/* Nhận xét:
+- Bài này anh bị overthinking rồi, anh chả biết nói gì nữa.
+*/
+
+class Solution {
+public:
+    typedef pair<int, int> pii;
+    typedef pair<int, pii> pip;
+    vector<int> powerfulIntegers(int x, int y, int bound) {
+        if (bound < 2) {
```

Comment:
Có thể code ngắn gọn hơn. Lược bỏ mấy phần if corner case này đi được. Bài này có thể code tương tự như bài hôm trước, chú có thể nhét hết (x^0, y^0), (x^1, y^0), (x^2, y^0), (x^3, y^0), (x^4, y^0)... vào heap. Tại mỗi bước lấy ra được (X^k, y^t) thì chỉ cần nhét (x^k, y^t + 1) vào thôi. Code sẽ ngắn gọn hơn.

Reply:
Anh bỏ corner cases đi rồi, thanks chú.
Ý tưởng của anh cũng giống hệt chú bảo đấy, đẩy hết một đầu vào heap, rồi cho chạy theo hướng còn lại. 
Tuy nhiên anh thấy bị overthinking vì bài này có thể giải đơn giản bằng brute force, liệt kê tất cả các cặp i, j. Mỗi chiều chỉ có log(bound) giá trị có thể (xấp xỉ 20) cho nên số nghiệm có thể rất ít (chỉ khoảng 400).
Trong khi bài này không yêu cầu liệt kê theo thứ tự, cũng ko cần chọn phần tử thứ k, nên dùng heap chả có tác dụng gì mà còn làm tăng độ phức tạp :(

Reply:
ừ nếu đề bài chỉ đơn giản, ko cần thứ tự thì làm for trâu cũng được. Cơ mà code heap cho nó chắc. :D